cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -86,9 +87,9 @@ ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-Wno-deprecated-declarations")
 
 IF(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   #
-  # Silence Clang warnings about unused parameters:
+  # Silence Clang warnings about unused parameters (mainly caused by BOOST):
   #
-  SET(DEAL_II_CXX_FLAGS "-Qunused-arguments ${DEAL_II_CXX_FLAGS}")
```

Comment:
The <code>-Qunused-arguments</code> does something else and is highly necessary for older clang version due to a regression.